query-parser/src/find.rs

```diff
@@ -38,10 +38,30 @@ extern crate mentat_query;
 
 use std::collections::BTreeMap;
 
-use self::mentat_query::{FindQuery, SrcVar};
-
-use super::error::{QueryParseError, QueryParseResult};
-use super::util::{values_to_variables, vec_to_keyword_map};
+use self::mentat_query::{
+    FindQuery,
+    FromValue,
+    SrcVar,
+    Variable,
+};
+
+use super::parse::{NotAVariableError, QueryParseError, QueryParseResult};
+use super::util::vec_to_keyword_map;
+
+/// If the provided slice of EDN values are all variables as
+/// defined by `value_to_variable`, return a Vec of Variables.
+/// Otherwise, return the unrecognized Value.
```

Comment:
nit: an `Error` wrapping the unrecognized `Value`?  Or don't wrap in `Err` at all (since you have set the precedent).